docs/user-guide/how_to_use_nodegroup_plugin.md

```diff
@@ -0,0 +1,104 @@
+# Nodegroup Plugin User Guide
+
+## Introduction
+
+**Nodegroup plugin** is designed to isolate resources by assigning labels to nodes. 
+
+## How the Nodegroup Plugin Works
+
+The nodegroup design document provides the most detailed information about the node group. There are some tips to help avoid certain issues.
+
+1. Soft constraints are a subset of hard constraints, including both affinity and anti-affinity. Consider a queue setup as follows:
```

Comment:
The following steps are both about queue's configuration, should also add other steps, like set node lable, and submit jobs to a queue and verfiy that jobs are scheduled to affinity nodes and schedule failed on antiaffinity nodes.